Tests/UITests/TestApp/LLMOpenAIRealtime/AudioViewModel.swift

```diff
@@ -0,0 +1,95 @@
+//
+// This source file is part of the Stanford Spezi open source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SpeziLLMOpenAIRealtime
+import SwiftUI
+
+@Observable
+@MainActor
+final class AudioViewModel {
+    private let replayUserAudio: Bool = false
+    
+    private var streamingService: AudioRecorder?
+    // Used to play pcm from user's microphone (playback & debugging)
+    private let pcmUserAudioPlayer = PCMPlayer()
+    // Used to play pcm from OpenAI's result
+    private let pcmOpenAiPlayer = PCMPlayer()
+
+    // The microphone --> LLM task
+    private var micTask: Task<Void, any Error>?
+    // The LLM --> PCMPlayer (audio) task
+    private var llmTask: Task<Void, any Error>?
+    
+    var isRecording: Bool = false
+    
+    func setup(llm: LLMOpenAIRealtimeSession) {
+        Task {
```

Comment:
We should generally aim to avoid creating Tasks across the codebase, only when we really need to introduce a concurrent construct to improve parallel execution, otherwise we should aim for structured concurrency flows across async functions.